doc/api/addons.md

```diff
@@ -98,6 +98,118 @@ the `.node` suffix).
 In the `hello.cc` example, then, the initialization function is `Initialize`
 and the addon module name is `addon`.
 
+When building addons with `node-gyp`, using the macro `NODE_GYP_MODULE_NAME` as
+the first parameter of `NODE_MODULE()` will ensure that the name of the final
+binary will be passed to `NODE_MODULE()`.
+
+### Context-aware addons
+
+There are environments in which Node.js addons may need to be loaded multiple
+times in multiple contexts. The macro `NODE_MODULE_INIT()` will construct an
```

Comment:
Can you please articulate what _could_ be those cases?

Reply:
@mcollina how about the new paragraph I added?